src/Build/Instance/TaskFactories/TaskHostTask.cs

```diff
@@ -206,7 +206,7 @@ public object GetPropertyValue(TaskPropertyInfo property)
 
                 // If we returned an exception, then we want to throw it when we 
                 // do the get.  
-                if (value != null && (value as Exception) != null)
+                if (value is Exception)
                 {
                     throw (Exception)value;
                 }
```

Comment:
nit: I know it's already merged. Just in case you're planning to submit more refactoring PRs, this could have been:

```C#
                if (value is Exception exception)
                {
                    throw exception;
                }
```

Thank you for the contributions!

Reply:
I plan on going through every refactor suggestion. This is one of them that is in the works :)